Tests/Unit/Hooks/HookProviderTest.php

```diff
@@ -0,0 +1,365 @@
+     */
+    public function hookObjectForNoInterfaceCannotBeCreated()
+    {
+        $this->expectException(\UnexpectedValueException::class);
+        $this->expectExceptionCode(1565089078);
+
+        $subject = new HookProvider('');
+    }
+
+    /**
+     * @test
+     */
+    public function hookObjectForNonexistantInterfaceCannotBeCreated()
+    {
+        $this->expectException(\UnexpectedValueException::class);
+        $this->expectExceptionCode(1565089078);
+
+        $subject = new HookProvider('\OliverKlee\Seminars\Tests\Unit\Fixtures\Hooks\TestingHookInterfaceDoesNotExist');
+    }
+
+    /**
+     * @test
+     */
+    public function hookObjectForInvalidInterfaceCannotBeCreated()
+    {
+        $this->expectException(\UnexpectedValueException::class);
+        $this->expectExceptionCode(1565088963);
+
+        $subject = new HookProvider(TestingHookInterfaceNotExtendsHook::class);
+    }
+
+    /**
+     * @test
+     * @doesNotPerformAssertions
+     */
+    public function hookObjectForTestHookWithNoHookImplementorRegisteredSucceedsForValidMethod()
+    {
+        unset($GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TestingHookInterface::class]);
+        $hookObject = $this->createHookObject();
+
+        $hookObject->executeHook('testHookMethod');
+    }
+
+    /**
+     * @test
+     */
+    public function hookObjectForTestHookWithNoHookImplementorRegisteredFailsForEmptyMethod()
+    {
+        unset($GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TestingHookInterface::class]);
+        $hookObject = $this->createHookObject();
+
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionCode(1573479911);
+
+        $hookObject->executeHook('');
+    }
+
+    /**
+     * @test
+     */
+    public function hookObjectForTestHookWithNoHookImplementorRegisteredFailsForUnknownMethod()
+    {
+        unset($GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TestingHookInterface::class]);
+        $hookObject = $this->createHookObject();
+
+        $this->expectException(\UnexpectedValueException::class);
+        $this->expectExceptionCode(1573480302);
+
+        $hookObject->executeHook('methodNotImplemented');
+    }
+
+    /**
+     * @test
+     */
+    public function hookObjectForTestHookWithOneHookImplementorRegisteredSucceedsWithMethodCalledOnce()
+    {
+        TestingHookImplementor::$wasCalled = 0;
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][TestingHookInterface::class][1565007112] =
+            TestingHookImplementor::class;
+        $hookObject = $this->createHookObject();
+
+        $hookObject->executeHook('testHookMethod');
+
+        self::assertSame(1, TestingHookImplementor::$wasCalled);
+    }
+
+    /**
+     * @test
+     */
+    public function hookObjectForTestHookWithTwoHookImplementorsRegisteredSucceedsWithEachMethodCalledOnce()
+    {
+        TestingHookImplementor::$wasCalled = 0;
+        TestingHookImplementor2::$wasCalled = 0;
```

Comment:
I propose to move this kind of pre-test cleanup into `setUp()` so each test always starts with a clean slate.